From 8acb3b594f304421c06001630f652d11246fa84a Mon Sep 17 00:00:00 2001
From: Simon Morlat <simon.morlat@linphone.org>
Date: Fri, 23 Mar 2018 16:24:33 +0100
Subject: [PATCH 2/2] Fix firebase (fix actually backported from dev_group_chat
 branch, with modifications to handle a build without the google-services
 jar.)

---
 src/android/org/linphone/LinphoneManager.java | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/src/android/org/linphone/LinphoneManager.java b/src/android/org/linphone/LinphoneManager.java
index 5e8a901e..53ecacda 100644
--- a/src/android/org/linphone/LinphoneManager.java
+++ b/src/android/org/linphone/LinphoneManager.java
@@ -691,10 +691,18 @@ public class LinphoneManager implements CoreListener, SensorEventListener, Accou
 				Log.i("[Push Notification] Assuming GCM jar is not provided.");
 			}
 		}else if (getString(R.string.push_type).equals("firebase")){
-			final String refreshedToken = com.google.firebase.iid.FirebaseInstanceId.getInstance().getToken();
-			if (refreshedToken != null) {
-				Log.i("[Push Notification] current token is: " + refreshedToken);
-				LinphonePreferences.instance().setPushNotificationRegistrationID(refreshedToken);
+			try{
+				Class<?> firebaseClass = Class.forName("com.google.firebase.iid.FirebaseInstanceId");
+				Object firebaseInstance = firebaseClass.getMethod("getInstance").invoke(null);
+				final String refreshedToken = (String)firebaseClass.getMethod("getToken").invoke(firebaseInstance);
+			
+				//final String refreshedToken = com.google.firebase.iid.FirebaseInstanceId.getInstance().getToken();
+				if (refreshedToken != null) {
+					Log.i("[Push Notification] current token is: " + refreshedToken);
+					LinphonePreferences.instance().setPushNotificationRegistrationID(refreshedToken);
+				}
+			}catch(Exception e){
+				Log.i("[Push Notification] firebase not available.");
 			}
 		}
 	}
-- 
2.11.0

